### PR TITLE
Add reference genome size to typing report

### PR DESCRIPTION
**Pull request by henningonsbring**
# Description
Adding a row to the typing report with the size of the reference genome given by the customer. This will allow a comparison between the assembly size and the size of the reference.

Also this PR fixes an error in the installation of microSALT.

## Primary function of PR
- [ ] Hotfix
- [ ] Patch
- [x] Minor functionality improvement
- [ ] New type of analysis
- [ ] Backward-breaking functionality improvement
- [ ] This change requires internal documents to be updated
- [ ] This change requires another repository to be updated

# Testing
_If the update is a hotfix, it is sufficient to rely on the development testing along with the Travis self-test automatically applied to the PR._

_Test routine to verify the stability of the PR:_
- _`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-microsalt-stage.sh genome-size-comp`_
- _`us`_
- _`source activate S_microSALT`_
- _(SITUATIONAL) `export MICROSALT_CONFIG=/home/proj/dropbox/microSALT.json`_
- _Select a relevant subset of the following:_
- _`microSALT analyse project MIC3109`_
- _`microSALT analyse project MIC4107`_
- _`microSALT analyse project MIC4109`_
- _`microSALT analyse project ACC5551`_
- _`cg workflow microsalt start -t 609108`_

_Verify that the results for projects MIC3109, MIC4107, MIC4109 & ACC5551 are consistent with the results attached to AMSystem doc 1490, Microbial_WGS.xlsx_

## Test results
_These are the results of the tests, and necessary conclusions, that prove the stability of the PR._

# Sign-offs
- [x] Code tested by @henningonsbring 
- [ ] Approved to run at Clinical-Genomics by @talnor

### CODE REVIEW

**Comment by henningonsbring** [comment]
"Referensens genomstorlek" is now displayed in the typing report (and it matches the size identified with the `wc -c` pipe):
Screen Shot 2021-09-28 at 15 25 25

**Comment by henningonsbring** [comment]
New column for reference genome length in `/home/proj/stage/microbial/meta/microsalt.db` added with the following script (using the following command: `python /home/henning.onsbring/db_scripts/add_column_sqlite.py -c reference_length -t samples -d /home/proj/stage/microbial/meta/microsalt.db -y INTEGER`)
```
# Add a column to a SQLite database
import sqlite3

from argparse import ArgumentParser

parser = ArgumentParser()

parser.add_argument("-c", "--column", dest="column",
                    help="Name of column",
                    metavar="<string>", required=True)
parser.add_argument("-t", "--table", dest="table",
                    help="Name of table",
                    metavar="<string>", required=True)
parser.add_argument("-d", "--database", dest="database",
                    help="Path to database",
                    metavar="<path>", required=True)
parser.add_argument("-y", "--type", dest="type",
                    help="What is stored in the column",
                    metavar="<string>", required=True)

args = parser.parse_args()

# Create a connection object
connection  = sqlite3.connect(args.database)

# Get a cursor
cursor = connection.cursor()

# Add the column to the table
addColumn = "ALTER TABLE %s ADD COLUMN %s %s" % (args.table, args.column, args.type)
cursor.execute(addColumn)

print("Added column %s to table %s with the type %s" % (args.column, args.table, args.type))

# close the database connection
connection.close()
```

**Comment by talnor** [comment]
One question about the python script. Does it add `None` values for existing samples in the database or how does it work? :)

**Comment by henningonsbring** [comment]
> One question about the python script. Does it add `None` values for existing samples in the database or how does it work? :)

The script creates an empty column, it does not add something there. It should be the default value, but I am not sure what the default value is. That can be checked by looking into the database `/home/proj/stage/microbial/meta/microsalt.db`, however, every time you run a microsalt analysis for a ticket you will overwrite what ever value is in that column.

File: install.sh [hunk 1]
```diff
@@ -52,14 +52,25 @@ done
...
-conda install -y -c bioconda -c conda-forge blast=2.12.0 bwa=0.7.17 picard=2.20.3 pigz=2.4 quast=5.0.2 samtools=1.13 spades=3.13.1 trimmomatic=0.39=1 r-base=4.1.1
+conda config --add channels conda-forge
+conda install -y blast=2.12.0
+conda install -y bwa=0.7.17
+conda install -y picard=2.20.3
+conda install -y pigz=2.4
+conda install -y quast=5.0.2
+conda install -y samtools=1.13
+conda install -y spades=3.13.1
+conda install -y trimmomatic=0.39
```
Comment (talnor) [inline install.sh:71]: Awesome! 👍 Can we for clarity also include `-c conda-forge` for r-base and `-c bioconda` for the other tools?

Comment (henningonsbring) [reply install.sh:71]: I can include it when you explain the difference between how I do it now and how you suggest we should do it. How we have it now is (as I understand it) how to install these channels, and it works.

**Comment by henningonsbring** [comment]
ran single sample: `cg workflow microsalt start -s ACC8231A4`
Screen Shot 2021-12-07 at 17 08 46

print screen show how reference genome size is displayed in the typing report after analysing single samples

**Comment by henningonsbring** [comment]
@talnor time to merge this? it is working now, both for tickets and single samples. Since the new PR is working, also obviously the install-script is working. I have also rephrased a warning message based on your review.

File: install.sh (hunk 1 above)
Comment (talnor) [inline install.sh:71]:
```suggestion
conda install -y trimmomatic=0.39=1
```

Comment (talnor) [reply install.sh:71]: or change path in init.

Comment (henningonsbring) [reply install.sh:71]: changed to this in init:
```
        preset_config["folders"]["adapters"] = os.path.abspath(
            os.path.join(
                os.path.expandvars("$CONDA_PREFIX"),
                "share/trimmomatic/adapters/",
            )
        )
```

**Review by talnor — APPROVED** [review]
Looks great! Awesome job!

**Comment by henningonsbring** [comment]
Ran on hasta:
```
python /home/henning.onsbring/db_scripts/add_column_sqlite.py -c reference_length -t samples -d /home/proj/production/microbial/meta/microsalt.db -y INTEGER
Added column reference_length to table samples with the type INTEGER
```

**Pull request merged**